bkg-ecosystem

HelenaLC opened the pull request:
- We had initially discussed (and discarded) the idea of giving a tabular overview of available methods (not maintainable)
- I thought it would be fun/interesting to use `biocViews` plus `BiocPkgTools` to do this programmatically
- To get started/learn, I demonstrate some simple operations, and then quantify
  - how the number of packages evolves over time
  - how many packages are added each year (on average)
  - how long packages remain available on Bioconductor
- Currently, this is limited to "SingleCell" and "Spatial" terms
- Next, I am considering to explore combinations of terms
  - "Spatial+Clustering", "+DimensionReduction", etc., over time
  - upset plot of the number of packages that share a (sub)set of terms
  - these numbers could also be rendered as a table using `kable`
- Any fun/interesting ideas anyone might have are welcome!
- The premise of the chapter is given in the introduction as:
> This chapter is not directly aimed at users, but aims at exploring the space of 
packages available through Bioconductor that revolve around spatial and single
cell omics data.

WARNING: This PR adds `Suggests: BiocPkgTools` and would require rebuilding the dependencies Docker.

lmweber commented:
I'd probably recommend shortening that first sentence and simply saying something like "This chapter provides additional background about the Bioconductor ecosystem by exploring the space of packages available through Bioconductor that revolve around spatial and single cell omics data." - otherwise I worry that readers might wonder why we are including it.

lmweber commented on inst/pages/bkg-ecosystem.qmd:
````diff
@@ -0,0 +1,179 @@
+# will be added in the next release
+```
+
+To get an idea of the projects' "growth" in this space, we can regress the 
````
typo: project's

HelenaLC commented:
Thanks for reviewing! I think I addressed the points you made. Also added another analysis on (co)occurence of subterms (e.g., clustering). For me this would be ok to merge, and can be revised at a later point in time. I like the idea that 6 months from now the plots will continue to "evolve" ;)

HelenaLC commented:
Ok wait... I messed sth up with my branches (recent commits are not showing). Will be on it & ping when ready...
